CHANGELOG.md

```diff
@@ -4,6 +4,10 @@
 
 Version <dev> brings support for gleaning a Docker container id from cgroups v2 based containers and records additional synthetics attributes.
 
+- **Feature: Prevent the agent from starting in additional Rails CLI/TUI contexts**
```

Comment:
maybe we should clarify that this fixes an issue on rails 7? since older rails versions already have this working the the rake task config?

Reply:
I thought about that. Unfortunately I learned that `bin/rails routes` isn't particularly new and has been around since at least Rails 5.0. So while it's true that the agent and its autostart denylist logic predates Rails 5.0 and that's why we never denylisted the `bin/rails` subcommands previously, it's likely that this change applies to the majority of current Rails shops.

Reply:
(From an offline conversation) I was incorrect in my understanding of the problem. Even though Rails 5 knew what to do with `rails routes`, the `routes` bit was actually being handed off to Rake.

- Beginning of Rails - had very limited rails binary functionality, use rake for everything except starting the server
- Version ? - the rails binary is now a one stop shop. Run rails foo and Rails will either know about foo and do something, or work as a pass-through to hand foo off to Rake
- Version 7 - certain subcommands such as routes are now known to Rails and handled by Rails commands instead of being handed off to Rake

So it's Rails 7 users who will benefit from this PR, with 5 and 6 users already having autostart disabled via Rake related denylist functionality.

Thanks for clearing things up for me, @tannalynn!